### ISSUE TITLE
list objects: Encoding type set to url is not honored

### ISSUE THREAD

**Issue by vadmeste** [minio/minio#7260]
## Expected Behavior
When listing objects is configured with encoding type set to "url", I expect that object names in listing response should be url encoded

## Current Behavior
Object names are not url encoded

## Possible Solution
Add encoding type feature

## Steps to Reproduce (for bugs)
Compare the behavior of the following code between Minio and AWS S3, after creating a bucket and uploading an object which has a space in its name (e.g. "file name")

1. `go get -u github.com/aws/aws-sdk-go`
2. 

```go
package main

import (
	"fmt"

	"github.com/aws/aws-sdk-go/aws"
	"github.com/aws/aws-sdk-go/aws/credentials"
	"github.com/aws/aws-sdk-go/aws/session"
	"github.com/aws/aws-sdk-go/service/s3"
)

func main() {
	sess := session.Must(session.NewSession())

	listingFn := func(p *s3.ListObjectsOutput, last bool) (shouldContinue bool) {
		for _, obj := range p.Contents {
			fmt.Println("Object:", *obj.Key)
		}
		return true
	}

	svc := s3.New(sess, &aws.Config{
		Credentials:      credentials.NewStaticCredentials("minio", "minio123", ""),
		Endpoint:         aws.String("localhost:9000"),
		Region:           aws.String("us-east-1"),
		S3ForcePathStyle: aws.Bool(true),
		DisableSSL:       aws.Bool(true),
	})

	listRequest := s3.ListObjectsInput{
		Bucket:       aws.String("vadmeste-test"),
		EncodingType: aws.String("url"),
	}

	err := svc.ListObjectsPages(&listRequest, listingFn)
	if err != nil {
		fmt.Println("failed to list objects", err)
		return
	}
}
```

## Context
Honor encoding type in object listing.

## Regression
No

## Your Environment
All

**Comment by harshavardhana** [comment]
Yep it's an old feature requirement which we never implemented. Is this related to XML parsing issue ?

**Comment by vadmeste** [comment]
> Yep it's an old feature requirement which we never implemented. Is this related to XML parsing issue ?

No, this is *probably* related to a crash in Vertica software, I compared Minio & AWS output and found this.

**Comment by harshavardhana** [comment]
Perhaps the parsing again, but Vertica shouldn't crash. AWS CLI also sends encoding-type as URL and never had any problems.

**Comment by vadmeste** [comment]
> Perhaps the parsing again, but Vertica shouldn't crash. AWS CLI also sends encoding-type as URL and never had any problems.

Yes, but probably Vertica logic is expecting something to have for sure as output and crash when it doesn't find it.

**Comment by harshavardhana** [comment]
Fixed by #7265
